vm/devices/storage/disklayer_sqlite/src/resolver.rs

```diff
@@ -41,17 +49,91 @@ impl ResolveResource<DiskLayerHandleKind, SqliteDiskLayerHandle> for SqliteDiskL
         }) = format_dbhd
         {
             ResolvedDiskLayer::new(FormatOnAttachSqliteDiskLayer::new(
-                dbhd_path,
+                dbhd_path.into(),
                 input.read_only,
                 crate::IncompleteFormatParams {
                     logically_read_only,
                     len,
                 },
-            )?)
+            ))
         } else {
-            ResolvedDiskLayer::new(SqliteDiskLayer::new(dbhd_path, input.read_only, None)?)
+            ResolvedDiskLayer::new(SqliteDiskLayer::new(
+                dbhd_path.as_ref(),
+                input.read_only,
+                None,
+            )?)
         };
 
         Ok(layer)
     }
 }
+
+impl ResolveResource<DiskLayerHandleKind, SqliteAutoCacheDiskLayerHandle>
+    for SqliteDiskLayerResolver
+{
+    type Output = ResolvedDiskLayer;
+    type Error = anyhow::Error;
+
+    fn resolve(
+        &self,
+        rsrc: SqliteAutoCacheDiskLayerHandle,
+        input: ResolveDiskLayerParameters<'_>,
+    ) -> Result<Self::Output, Self::Error> {
+        let layer = ResolvedDiskLayer::new(AutoCacheSqliteDiskLayer {
+            path: rsrc.cache_path.into(),
+            key: rsrc.cache_key,
+            read_only: input.read_only,
+        });
+        Ok(layer)
+    }
+}
+
+struct AutoCacheSqliteDiskLayer {
```

Comment:
this struct + impl really feel like they should be in their own mod (vs. just hanging out here in `resolver.rs`)